compiler/rustc_lint/src/context/diagnostics/check_cfg.rs

```diff
@@ -41,6 +41,25 @@ fn check_cfg_expected_note(
     note
 }
 
+enum EscapeQuotes {
+    Yes,
+    No,
+}
+
+fn to_check_cfg_arg(name: Symbol, value: Option<Symbol>, quotes: EscapeQuotes) -> String {
+    if let Some(value) = value {
+        let values = match quotes {
+            EscapeQuotes::Yes => {
+                format!("\\\"{}\\\"", str::escape_debug(value.as_str()).to_string())
+            }
+            EscapeQuotes::No => format!("\"{value}\""),
```

Comment:
Pretty sure this escaping, too: `#[cfg(key = "value\"")]`. Current suggestion: `--check-cfg=cfg(key, values("value""))` I think

Reply:
Indeed, fixed and added a test.